[fx] Don't use __module__ to test if a function is bound from C++

peterbell10 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #73434
* #73433
* __->__ #75896

The new `test_public_bindings.py` test means `__module__` will be set
correctly in future, even for functions bound from C++. Instead, just
test directly that the function is of the `BuiltinFunctionType` which
only passes for functions exported with the CPython API.

peterbell10 committed:
[fx] Don't use __module__ to test if a function is bound from C++

The new `test_public_bindings.py` test means `__module__` will be set
correctly in future, even for functions bound from C++. Instead, just
test directly that the function is of the `BuiltinFunctionType` which
only passes for functions exported with the CPython API.

[ghstack-poisoned]

peterbell10 committed:
Update on "[fx] Don't use __module__ to test if a function is bound from C++"


The new `test_public_bindings.py` test means `__module__` will be set
correctly in future, even for functions bound from C++. Instead, just
test directly that the function is of the `BuiltinFunctionType` which
only passes for functions exported with the CPython API.

[ghstack-poisoned]

peterbell10 committed:
Update on "[fx] Don't use __module__ to test if a function is bound from C++"


The new `test_public_bindings.py` test means `__module__` will be set
correctly in future, even for functions bound from C++. Instead, just
test directly that the function is of the `BuiltinFunctionType` which
only passes for functions exported with the CPython API.

[ghstack-poisoned]

albanD reviewed (approved):
SGTM

albanD commented:
@pytorchbot merge this please

albanD commented:
cc @peterbell10 looks like this needs a rebase!
Feel free to land once CI is green after that though!

peterbell10 committed:
Update on "[fx] Don't use __module__ to test if a function is bound from C++"


The new `test_public_bindings.py` test means `__module__` will be set
correctly in future, even for functions bound from C++. Instead, just
test directly that the function is of the `BuiltinFunctionType` which
only passes for functions exported with the CPython API.

[ghstack-poisoned]

peterbell10 committed:
Update on "[fx] Don't use __module__ to test if a function is bound from C++"


The new `test_public_bindings.py` test means `__module__` will be set
correctly in future, even for functions bound from C++. Instead, just
test directly that the function is of the `BuiltinFunctionType` which
only passes for functions exported with the CPython API.

[ghstack-poisoned]

peterbell10 commented:
@pytorchbot merge this
